pkg/option/config.go

```diff
@@ -3123,6 +3155,59 @@ func (c *DaemonConfig) calculateDynamicBPFMapSizes(totalMemory uint64, dynamicSi
 	}
 }
 
+// Validate VTEP support configuration
+func (c *DaemonConfig) validateVTEP() error {
+	vtepEndpoints := viper.GetStringSlice(VtepEndpoint)
+	vtepCIDRs := viper.GetStringSlice(VtepCIDR)
+	vtepMACs := viper.GetStringSlice(VtepMAC)
+
+	if (len(vtepEndpoints) < 1) ||
+		len(vtepEndpoints) != len(vtepCIDRs) ||
+		len(vtepEndpoints) != len(vtepMACs) ||
+		len(vtepCIDRs) != len(vtepMACs) {
+		err := errors.New("VTEP config number does not match")
+		log.WithError(err).WithFields(logrus.Fields{
+			"vtep": len(vtepEndpoints),
+		}).Error("Endpoints or CIDRs or MACs number does not match")
+		return err
```

Comment:
I think that all of these patterns where the code is doing log + return error should either log OR return the error. Note that returning the error is how we ensure that the agent rejects the bad configurations.

For this first one it's a little tricky to describe, here's an example:

```suggestion
		return fmt.Errorf("VTEP configuration must have the same number of Endpoint, VTEP and MAC configurations (Found %d endpoints, %d MACs, %d CIDR ranges)", len(vtepEndpoints), len(vtepMACs), len(vtepCIDRS)"
```

Reply:
Thanks, done